Story/DageTheEvilIsland/CoreDageTheEvilIsland.cs

```diff
@@ -0,0 +1,315 @@
+
+        //Defeat Vincenzo! 4194
+        Story.KillQuest(4194, "LegionCrypt", "Vincenzo");
+
+        //Defeat Brutus! 4195
+        Story.KillQuest(4195, "LegionCrypt", "Brutus");
+    }
+
+    public string[] RequiredItems = { "Envy Token V", "Envy Token IV", "Envy Token III", "Envy Token II", "Envy Token I" };
+
+    public void EnvyMap()
+    {
+        if (Core.isCompletedBefore(4893))
+        {
+            Core.Logger("You have already completed Envy storyline");
+            return;
+        }
+        Story.PreLoad();
+
+        Core.AddDrop(RequiredItems);
+
+        if (!Bot.Quests.IsUnlocked(4891))
+        {
+            if (!Core.CheckInventory("Envy Token V"))
+            {
+                if (!Core.CheckInventory("Envy Token IV"))
+                {
+                    if (!Core.CheckInventory("Envy Token III"))
+                    {
+                        if (!Core.CheckInventory("Envy Token II"))
+                        {
+                            if (!Core.CheckInventory("Envy Token I"))
+                            {
+                                //An Enviable Army 4884
+                                Core.EnsureAccept(4884);
+                                Core.HuntMonster( "Envy", "Legion Defector", "Legion Defectors Beaten", 3 );
+                                Core.HuntMonster( "Envy", "Fawning Sycophant", "Fawning Sycophants Beaten", 4 );
+                                Core.EnsureComplete(4884);
+                                Bot.Wait.ForPickup("Envy Token I");
+                            }
+                            //Where’s Darkon? 4885
+                            Core.EnsureAccept(4885);
+                            Core.GetMapItem(4278, 4, "Envy");
+                            Core.EnsureComplete(4885);
+                            Bot.Wait.ForPickup("Envy Token II");
+                        }
+                        //Her Biggest Fans 4886
+                        Core.EnsureAccept(4886);
+                        while (!Bot.TempInv.Contains("Sycophant Hood"))
+                            Core.HuntMonster("Envy", "Fawning Sycophant", "Sycophant Hood");
+                        while (!Bot.TempInv.Contains("Sycophant Tunic"))
+                            Core.HuntMonster("Envy", "Fawning Sycophant", "Sycophant Tunic");
+                        while (!Bot.TempInv.Contains("Sycophant Medallion"))
+                            Core.HuntMonster("Envy", "Fawning Sycophant", "Sycophant Medallion");
+                        while (!Bot.TempInv.Contains("Sycophant Boots"))
+                            Core.HuntMonster("Envy", "Fawning Sycophant", "Sycophant Boots");
+                        Core.EnsureComplete(4886);
+                        Bot.Wait.ForPickup("Envy Token III");
+                    }
+                    //Squee for Envy 4887
+                    Core.EnsureAccept(4887);
+                    while (!Bot.TempInv.Contains("Clue"))
```

Comment:
the While()'s here arent needed, just do a Core.HuntMonster here, the default on Core.HuntMonster monster is already IsTemp: True so your fine 👍  same goes for the rest of these in this section, and for the future, if the item isn't temp same thing just do isTemp: false,